cross_domain_composability_protocol.md

```diff
@@ -0,0 +1,804 @@
+7. In case there's a timeout (before a `Vote` message has been sent) or if the transaction simulation fails but not due to a mailbox read error, it sends a `Vote(0)` message to the SP, indicating the impossibility of including the transaction, and terminates.
+8. In case a `Decided(1)` message is received from the SP, it adds all `mailbox.putInbox' transactions created and the main transaction to the block, and terminates.
+9. In case a `Decided(0)` message is received from the SP, it removes all `mailbox.putInbox' transactions created and the main transaction from the block, reverting to the previous state, and terminates.
+
+The Shared Publisher (SP) runs a slightly modified version of the SCP protocol:
+1. It sends a `StartCDCP` message to the appropriate sequencers, starts a timer and waits for NSs' `Vote` messages.
+2. If a timeout occurs or if a `Vote(0)` message is received (indicates a failure), it sends a `Decided(0)` message to the NSs and a `NativeDecided(0)` message to the WS, and terminates.
+3. If it receives a `Vote(1)` message from all NSs, it stops the timer (as it not longer will be used) and sends a `NativeDecided(1)` message to the WS, indicating that the NSs are willing to proceed.
+4. Once a `WSDecided(b)` message is received from the WS, it sends a `Decided(b)` message indicating the result to the NSs, and terminates.
+
+Note that, from the above, the SP works as a middle layer between the NSs and WS.
+It indicates the viability results from the NSs to the WS, who is now allowed to communicate with the external rollup for including its transaction.
+Once the WS receives a response, it indicates the result to the SP, who transmits the result back to the NSs.
+
+The Wrapped Sequencer (WS) has the following rules:
+1. Same as NS.
+2. Same as NS.
+3. Whenever a `stagedMailbox.Write` operation is intercepted and fails:
+  - It sends a `Mailbox` message to the counterparty chain sequencer.
+  - It creates a `stagedMailbox.putOutbox` transaction and adds it locally to the transaction list, placing it before the main transaction.
+  - Then, it re-starts the transaction simulation going back to step 2.
+4. Same as NS.
+5. Same as NS.
+6. In case the transaction simulation is successful, it waits for a `NativeDecided` message from the SP.
+7. In case there's a timeout (before a `WSDecided` message has been sent) or if the transaction simulation fails but not due to a mailbox write or read error, it sends a `WSDecided(0)` message to the SP, indicating failure and terminates.
+8. If a `NativeDecided(0)` message is received from the SP, it removes its transaction (and any created `stagedMailbox`) and terminates.
+9. If a `NativeDecided(1)` message is received from the SP and its local transaction simulation is successful, it stops the timer (as it no longer will be used) and sends a special transaction to the ER, which populates the mailbox and executes the transaction atomically.
```

Comment:
you meant a bundle of transactions? not sure what it means to execute a single "transaction atomically"